develop/api/CommunicationGatewayErrorCodes.md

```diff
@@ -0,0 +1,84 @@
+|GnmiHandler_RegisterSubscriptionRequest_InvalidSubscriptionName|gNMI handler Register subscription received invalid subscription name|1002003|The RegisterSubscriptionRequest contained a SubscriptionId with a SubscriptionName that can not be used as a message broker subject token.|
+|GnmiHandler_RegisterSubscriptionRequest_DuplicateSubscriptionId|gNMI handler Register subscription received duplicate subscription ID|1002004|The RegisterSubscriptionRequest contained a SubscriptionId that is already registered. No further actions were taken. Sending a SubscribeRequest for this SubscriptionId may interfere with an existing subscription.|
+|GnmiHandler_UnregisterSubscriptionRequest_UnknownSubscriptionId|gNMI handler Register subscription received an unknown subscription ID|1002005|The UnregisterSubscriptionRequest contained a SubscriptionId that is not known. Either it was already unregistered, or the wrong SubscriptionId was provided.|
+|GnmiHandler_GetRequest_InvalidMessage|gNMI handler invalid GetRequest|1002006|The GetRequest could not be parsed, this topic should send a Request message containing a GetRequest.|
+|GnmiHandler_GetRequest_UnexpectedFail|gNMI handler Get failed without status code|1002007|The GetAsync call threw an exception from which an error code could not be extracted. Check the CommunicationGateway logging for more information.|
+|GnmiHandler_SetRequest_InvalidMessage|gNMI handler invalid SetRequest|1002008|The SetRequest could not be parsed, this topic should send a Request message containing a SetRequest.|
+|GnmiHandler_SetRequest_UnexpectedFail|gNMI handler Set failed without status code|1002009|The SetAsync call threw an exception from which an error code could not be extracted. Check the CommunicationGateway logging for more information.|
+|GnmiHandler_CapabilityRequest_InvalidMessage|gNMI handler invalid CapabilityRequest|1002010|The SetRequest could not be parsed, this topic should send a Request message containing a CapabilityRequest.|
+|GnmiHandler_CapabilityRequest_UnexpectedFail|gNMI handler Capabilities failed without status code|1002011|The CapabilitiesAsync call threw an exception from which an error code could not be extracted. Check the CommunicationGateway logging for more information.|
+|GnmiHandler_RegisterSubscriptionRequest_InvalidMessage     |gNMI handler invalid RegisterSubscriptionRequest|1002012|The RegisterSubscriptionRequest could not be parsed, this topic should send a Request message containing a RegisterSubscriptionRequest.|
+|GnmiHandler_RegisterSubscriptionRequest_TryAddFailed|gNMI handler Register subscription TryAdd failed|1002013|The RegisterSubscriptionRequest contained a SubscriptionId that could not be added after checking for duplicate IDs. This can happen when sending the same request in parallel.|
```

Comment:
```suggestion
|GnmiHandler_RegisterSubscriptionRequest_TryAddFailed|gNMI handler Register subscription TryAdd failed|1002013|The RegisterSubscriptionRequest contained a SubscriptionId that could not be added after checking for duplicate IDs. This can happen when the same request is sent in parallel.|
```